File: Backend-.NetCore/SharedLayer/Dtos/OrderDto.cs
namespace SharedLayer.Dtos
{
    public class OrderDto
    {
        public string Id { get; set; }
        public string Date { get; set; }
        public string Status { get; set; } // 'Shipped', 'Delivered', 'Cancelled'
        public string Total { get; set; }
        public List<OrderedProductDto>? Products { get; set; } // Allow Products to be null
    }
}

File: FrontEnd-angular/src/app/Dtos/order.dto.ts
import { OrderedProductDto } from './orderedProduct.dto';

// we are using interfaces here because they are better than class 
// we used previously
export interface OrderDto {
  id: string;
  date: string;
  status: 'Shipped' | 'Delivered' | 'Cancelled';
  total: string;
  products: OrderedProductDto[];
}
